File: Token/src/main/java/com/aliucord/plugins/Token.java
package com.aliucord.plugins;

import android.content.Context;

import androidx.annotation.NonNull;

import com.aliucord.api.CommandsAPI;
import com.aliucord.entities.Plugin;
import com.aliucord.utils.ReflectUtils;
import com.discord.stores.StoreStream;

import java.util.Collections;

public class Token extends Plugin {
    @NonNull
    @Override
    public Manifest getManifest() {
        Manifest manifest = new Manifest();
        manifest.authors = new Manifest.Author[]{ new Manifest.Author("zt", 289556910426816513L) };
        manifest.description = "Adds a token slash command to tell you your Discord token.";
        manifest.version = "1.0.1";
        manifest.updateUrl = "https://raw.githubusercontent.com/zt64/aliucord-plugins/builds/updater.json";
        return manifest;
    }

    @Override
    public void start(Context context) {
        commands.registerCommand("token", "Tells you your token", Collections.emptyList(), ctx -> {
            try {
                String token = (String) ReflectUtils.getField(StoreStream.getAuthentication(), "authToken");
                return new CommandsAPI.CommandResult("```\n" + token + "```", null, false);
            } catch (NoSuchFieldException | IllegalAccessException e) {
                e.printStackTrace();
                return new CommandsAPI.CommandResult("Uh oh, failed to get token", null, false);
            }
        });
    }

    @Override
    public void stop(Context context) { commands.unregisterAll(); }
}

File: Token/src/main/java/com/aliucord/plugins/Token.kt
package com.aliucord.plugins

import android.content.Context
import com.aliucord.annotations.AliucordPlugin
import com.aliucord.api.CommandsAPI.CommandResult
import com.aliucord.entities.Plugin
import com.aliucord.utils.ReflectUtils
import com.discord.api.commands.ApplicationCommandType
import com.discord.models.commands.ApplicationCommandOption
import com.discord.stores.StoreStream
import java.util.*

@AliucordPlugin
class Token : Plugin() {
    override fun start(context: Context) {
        commands.registerCommand("token", "Tells you your token", Collections.singletonList(
            ApplicationCommandOption(ApplicationCommandType.BOOLEAN, "send", "Whether the result should be visible for everyone", null, false, false, null, null)
        )) {
            try {
                val token = ReflectUtils.getField(StoreStream.getAuthentication(), "authToken") as String?
                return@registerCommand CommandResult("```\n$token```", null, it.getBoolOrDefault("send", false))
            } catch (e: NoSuchFieldException) {
                e.printStackTrace()
                return@registerCommand CommandResult("Uh oh, failed to get token", null, false)
            } catch (e: IllegalAccessException) {
                e.printStackTrace()
                return@registerCommand CommandResult("Uh oh, failed to get token", null, false)
            }
        }
    }

    override fun stop(context: Context) = commands.unregisterAll()
}
